### ISSUE TITLE
[Android] Buttons are clickable when React Native Inspector is opened

### ISSUE THREAD

**Issue by angly-cat** [kmagiera/react-native-gesture-handler#218]
To date on Android if you open React Native Inspector and try to select a RNGH-button, that button would execute it's click action, making debugging rather inconvenient. On iOS inspector works as expected.

**Comment by osdnk** [comment]
Hi, @angly-cat 
Thank you a lot for this issue, however, I'm not able to fix it without changes in react-native code. Trigering inspectind has no side effects and then I'n not able to hook it. Switching inspector, does not imply any rerender or any callback. What's more, I found that as it's is possible to block touches on Touchables components, it's not possible to block them in our Library.
Am I right, @kmagiera?

**Comment by osdnk** [comment]
I decided to close this issue. I understand that problem is present but as it's really minor we are going to have radar on it later

**Comment by angly-cat** [comment]
Hi, @osdnk! Thanks for responding!

> I understand that problem is present but as it's really minor we are going to have radar on it later

This "clicking through" is very inconvenient when debugging RNGH's buttons' and handlers' styles/hitboxes etc.

> Trigering inspectind has no side effects and then I'n not able to hook it. Switching inspector, does not imply any rerender or any callback.

I've dived into `react-native`'s code and indeed found no ways to hook on inspector in `java` code. However, it's possible in JS-thread (found in [Libraries/ReactNative/AppContainer.js](https://github.com/facebook/react-native/blob/e7391438093cd5dd5033204cdce62e66509e66e1/Libraries/ReactNative/AppContainer.js#L58))

```
require('RCTDeviceEventEmitter').addListener(
    'toggleElementInspector',
    () => { /* React Native Inspector is toggled */ }
);
```

With that, if fixing this can be done in JS without putting too much effort, could you give this a second thought, please?

**Comment by osdnk** [comment]
Ok, but I'm not convinced is it a good idea to use is. Maybe it is good idea to make PR for react native to export some constant with "inspect enabled" state

**Comment by angly-cat** [comment]
> but I'm not convinced is it a good idea to use is

@osdnk I don't think that hooking to inspector toggling event is the best idea too. Therefore I dug into react-native's sources to find out how they "disable" touchables, so we could think of a non-hacky solution.

Apparently, when Inspector is toggled, [react-native cover the entire screen with `<View onStartShouldSetResponder={...}>`-overlay](https://github.com/facebook/react-native/blob/e7391438093cd5dd5033204cdce62e66509e66e1/Libraries/Inspector/InspectorOverlay.js#L75) for intercepting all touches. `onStartShouldSetResponder` returns `true`, so this overlay captures all touches before they could reach underneath touchables.

As I didn't develop RNGH I can't say for sure, but it seems that *by design* RNGH handles gestures before react-native's original gesture handling system, which causes this issue in the first place. However, since that's by design, this issue can only be fixed by somewhat hacky solutions like listening to Instector toggling events.

**Comment by osdnk** [comment]
You're right. Maybe it's better than nothing.
